crates/saya/core/src/lib.rs

```diff
@@ -142,6 +165,21 @@ impl Saya {
 
         let _exec_infos = self.provider.fetch_transactions_executions(block_number).await?;
 
+        let to_prove = ProvedStateDiff {
+            genesis_state_hash,
+            prev_state_hash: prev_block.header.header.state_root,
+            state_updates: state_updates_to_prove,
+        };
+
+        trace!(block_number, "proving block");
```

Comment:
As a refacto is on-going in an other PR, it could be great to use structured logging  in the new one being added. :+1:
(examples can be found [here](https://github.com/dojoengine/dojo/pull/1702))